scarb/src/compiler/plugin/proc_macro/host.rs

```diff
@@ -155,6 +156,186 @@ impl ProcMacroHostPlugin {
         })
     }
 
+    fn expand_inner_attr(
+        &self,
+        db: &dyn SyntaxGroup,
+        item_ast: ast::ModuleItem,
+    ) -> InnerAttrExpansionResult {
+        let mut context = InnerAttrExpansionContext::new(self);
+        let mut item_builder = PatchBuilder::new(db, &item_ast);
+
+        match item_ast.clone() {
+            ast::ModuleItem::Trait(trait_ast) => {
```

Comment:
nit: trait and impl logic are very similar. Maybe we could move some of the logic to separate function, so we just don't copy the code. Treat it as just a suggestion, i don't have a strong opinion on that

Reply:
Good point, done.